packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeForkClient.ts

```diff
@@ -57,6 +63,13 @@ Please use block number ${lastSafeBlock} or wait for the block to get ${
     forkBlockNumber = new BN(lastSafeBlock);
   }
 
+  const block = (await provider.request({
```

Comment:
Let's move this to a function, similar to how `getNetworkId` and `getLatestBlockNumber` are done.

Reply:
Ugh, forgot my own comment here. But this is a low-risk change, so I think this PR can still be reviewed.